Origin: r856 - 858
Description: update before*.rules to add safe icmp to ufw-before-forward,
 update doc/ufw.8 to describe the defaults, update section on default policy
 in README

Index: ufw-0.34~rc/conf/before6.rules
===================================================================
--- ufw-0.34~rc.orig/conf/before6.rules	2014-02-20 14:17:17.000000000 -0600
+++ ufw-0.34~rc/conf/before6.rules	2014-02-28 08:55:52.064702418 -0600
@@ -46,13 +46,20 @@
 -A ufw6-before-input -m conntrack --ctstate INVALID -j ufw6-logging-deny
 -A ufw6-before-input -m conntrack --ctstate INVALID -j DROP
 
-# ok icmp codes
+# ok icmp codes for INPUT
 -A ufw6-before-input -p icmpv6 --icmpv6-type destination-unreachable -j ACCEPT
 -A ufw6-before-input -p icmpv6 --icmpv6-type packet-too-big -j ACCEPT
 -A ufw6-before-input -p icmpv6 --icmpv6-type time-exceeded -j ACCEPT
 -A ufw6-before-input -p icmpv6 --icmpv6-type parameter-problem -j ACCEPT
 -A ufw6-before-input -p icmpv6 --icmpv6-type echo-request -j ACCEPT
 
+# ok icmp code for FORWARD
+-A ufw6-before-forward -p icmpv6 --icmpv6-type destination-unreachable -j ACCEPT
+-A ufw6-before-forward -p icmpv6 --icmpv6-type packet-too-big -j ACCEPT
+-A ufw6-before-forward -p icmpv6 --icmpv6-type time-exceeded -j ACCEPT
+-A ufw6-before-forward -p icmpv6 --icmpv6-type parameter-problem -j ACCEPT
+-A ufw6-before-forward -p icmpv6 --icmpv6-type echo-request -j ACCEPT
+
 # allow dhcp client to work
 -A ufw6-before-input -p udp -s fe80::/10 --sport 547 -d fe80::/10 --dport 546 -j ACCEPT
 
Index: ufw-0.34~rc/conf/before.rules
===================================================================
--- ufw-0.34~rc.orig/conf/before.rules	2014-02-20 14:17:17.000000000 -0600
+++ ufw-0.34~rc/conf/before.rules	2014-02-28 08:55:52.064702418 -0600
@@ -30,13 +30,20 @@
 -A ufw-before-input -m conntrack --ctstate INVALID -j ufw-logging-deny
 -A ufw-before-input -m conntrack --ctstate INVALID -j DROP
 
-# ok icmp codes
+# ok icmp codes for INPUT
 -A ufw-before-input -p icmp --icmp-type destination-unreachable -j ACCEPT
 -A ufw-before-input -p icmp --icmp-type source-quench -j ACCEPT
 -A ufw-before-input -p icmp --icmp-type time-exceeded -j ACCEPT
 -A ufw-before-input -p icmp --icmp-type parameter-problem -j ACCEPT
 -A ufw-before-input -p icmp --icmp-type echo-request -j ACCEPT
 
+# ok icmp code for FORWARD
+-A ufw-before-forward -p icmp --icmp-type destination-unreachable -j ACCEPT
+-A ufw-before-forward -p icmp --icmp-type source-quench -j ACCEPT
+-A ufw-before-forward -p icmp --icmp-type time-exceeded -j ACCEPT
+-A ufw-before-forward -p icmp --icmp-type parameter-problem -j ACCEPT
+-A ufw-before-forward -p icmp --icmp-type echo-request -j ACCEPT
+
 # allow dhcp client to work
 -A ufw-before-input -p udp --sport 67 --dport 68 -j ACCEPT
 
Index: ufw-0.34~rc/doc/ufw.8
===================================================================
--- ufw-0.34~rc.orig/doc/ufw.8	2014-02-20 14:17:17.000000000 -0600
+++ ufw-0.34~rc/doc/ufw.8	2014-02-28 08:55:52.064702418 -0600
@@ -431,7 +431,27 @@
 .PP
 On installation, \fBufw\fR is disabled with a default incoming policy of deny,
 a default forward policy of deny, and a default outgoing policy of allow, with
-stateful tracking for NEW connections.
+stateful tracking for NEW connections for incoming and forwarded connections.
+In addition to the above, a default ruleset is put in place that does the
+following:
+.TP
+- DROP packets with RH0 headers
+.TP
+- DROP INVALID packets
+.TP
+- ACCEPT certain icmp packets (INPUT and FORWARD): destination-unreachable, source-quench, time-exceeded, parameter-problem, and echo-request for IPv4. destination-unreachable, packet-too-big, time-exceeded, parameter-problem, and echo-request for IPv6.
+.TP
+- ACCEPT icmpv6 packets for stateless autoconfiguration (INPUT)
+.TP
+- ACCEPT ping replies from IPv6 link-local (ffe8::/10) addresses (INPUT)
+.TP
+- ACCEPT DHCP client traffic (INPUT)
+.TP
+- DROP non-local traffic (INPUT)
+.TP
+- ACCEPT mDNS (zeroconf/bonjour/avahi 224.0.0.251 for IPv4 and ff02::fb for IPv6) for service discovery (INPUT)
+.TP
+- ACCEPT UPnP (239.255.255.250 for IPv4 and ff02::f for IPv6) for service discovery (INPUT)
 
 .PP
 Rule ordering is important and the first match wins. Therefore when adding
@@ -439,7 +459,7 @@
 .PP
 \fBufw\fR is not intended to provide complete firewall functionality via
 its command interface, but instead provides an easy way to add or remove
-simple rules. It is currently mainly used for host\-based firewalls.
+simple rules.
 .PP
 The status command shows basic information about the state of the firewall, as
 well as rules managed via the \fBufw\fR command. It does not show rules from the
@@ -480,8 +500,9 @@
   ufw allow to 10.0.0.1 from 10.4.0.0/16 proto ah
 .PP
 In addition to the command\-line interface, \fBufw\fR also provides a
-framework which allows administrators to take full advantage of netfilter.
-See the \fBufw\-framework\fR manual page for more information.
+framework which allows administrators to modify default behavior as well as
+take full advantage of netfilter. See the \fBufw\-framework\fR manual page for
+more information.
 
 .SH SEE ALSO
 .PP
Index: ufw-0.34~rc/README
===================================================================
--- ufw-0.34~rc.orig/README	2014-02-20 14:17:17.000000000 -0600
+++ ufw-0.34~rc/README	2014-02-28 08:55:52.064702418 -0600
@@ -239,22 +239,19 @@
 - ACCEPT all RELATED and ESTABLISHED on FORWARD (ip forwarding must be enabled
   via sysctl for this to be in effect)
 - DROP INVALID packets (packets not associated with a known connection)
-- ACCEPT certain icmp packets:
+- ACCEPT certain icmp packets (INPUT and FORWARD):
   - destination-unreachable, source-quench, time-exceeded, parameter-problem,
     and echo-request for IPv4
-  - neighbor-solicitation, neighbor-advertisement, router-solicitation,
-    destination-unreachable, packet-too-big, time-exceeded, parameter-problem,
+  - destination-unreachable, packet-too-big, time-exceeded, parameter-problem,
     and echo-request
+- ACCEPT certain icmpv6 packets for stateless autoconfiguration (INPUT):
+  neighbor-solicitation, neighbor-advertisement, router-solicitation
 - ACCEPT mDNS (zeroconf/bonjour/avahi 224.0.0.251 for IPv4 and ff02::fb for
-  IPv6) for service discovery
+  IPv6) for service discovery (INPUT)
 - ACCEPT UPnP (239.255.255.250 for IPv4 and ff02::f for IPv6) for service
-  discovery
-- ACCEPT ping replies from IPv6 link-local (ffe8::/10) addresses
-- DROP non-local, broadcast and multicast traffic
-- ACCEPT DHCP client traffic
-- Silently DROP SMB/CIFS traffic
-- Silently DROP DHCP traffic not associated with host's use of DHCP client
-- Silently DROP BROADCAST (IPv4) traffic
+  discovery (INPUT)
+- ACCEPT ping replies from IPv6 link-local (ffe8::/10) addresses (INPUT)
+- ACCEPT DHCP client traffic (INPUT)
 - Log all blocked packets not matching the default policy with rate limiting
 
 If you are using a packaged version of ufw supplied by your distribution, the
